### DISCUSSION TITLE
Studio One Window Offset Issue on Hyprland

### DISCUSSION THREAD

**Discussion by Deecellar** [hyprwm/Hyprland#8743] [Bugs - Input]
# Studio One Window Offset Issue on Hyprland

I do music production as a hobby using Studio One, which recently released a Linux version. I'm experiencing an issue where the window has an incorrect offset that persists (except briefly during program startup and when resizing, where the navbar elements temporarily align correctly).

I'm using a slightly modified version of [This configuration](https://github.com/end-4/dots-hyprland) adapted for my machine. The modifications are mostly program-specific and don't affect the base Hyprland configurations.

Currently it looks like this:
![Window offset issue](https://github.com/user-attachments/assets/2ba36641-2e11-4589-b970-31925c3870ab)

Using the program's debug tools, I obtained the following Wayland log:
[studioone.txt](https://github.com/user-attachments/files/18158073/studioone.txt)

This is the registered offset (I use Hyprcursor, but I believe the app doesn't support it - it's a pure Wayland program with likely custom rendering):

https://github.com/user-attachments/assets/71a07fb1-f8f9-46db-842d-13aca2d0ae56

Before opening a bug report with Studio One, I want to investigate if this can be fixed from the Hyprland side, especially since the program works reasonably well on Ubuntu (with only standard program-specific bugs).

**Comment by jwolffpsl** [comment]
We use negative coordinates in xdg_surface::set_window_geometry, which might be a bit unusual, but is compliant with the spec.

See https://wayland.app/protocols/xdg-shell#xdg_surface:request:set_window_geometry: "The arguments are given in the surface-local coordinate space of the wl_surface associated with this xdg_surface, and may extend outside of the wl_surface itself to mark parts of the subsurface tree as part of the window geometry."

Could be related to #6682

**Comment by vaxerski** [comment]
yes this is kinda cursed if you do that, can open an issue

**Comment by norveganon** [comment]
I have the same issue. Will it be fixed?

**Comment by daigonstar** [comment]
Im having the same problem, dont wanna install another WM just for this app

**Comment by norveganon** [comment]
The offset seem to be about the same as the height of the option menu. Is it the the lack of window decoration in hyprland that is causing this problem? Look how the menu is above the window here: https://imgur.com/a/LYY3VNf

**Comment by ghost** [comment]
Any news on this one? I'm still having this problem with Studio One 7.2.3 and hyprland 0.52.2 on CachyOS

**Reply by gulafaran** [reply]
got a WAYLAND_DEBUG=1 log of running it?

**Comment by ghost** [comment]
Yes, I hope this helps.
[studioone.log](https://github.com/user-attachments/files/24070908/studioone.log)

**Comment by sean-phyx** [comment]
Also having this issue. I wish I could fix it and PR. Anyone got anything at all?

**Reply by norveganon** [reply]
I ended up switching to niri as my 'fix'.
